packages/eslint-plugin-react-native/src/rules/i18n-avoid-global-imports/index.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  AST_NODE_TYPES,
+  Identifier,
+  JSXMemberExpression,
+} from "@typescript-eslint/types/dist/generated/ast-spec";
+import { ESLintUtils } from "@typescript-eslint/utils";
+
+import { MessageIds } from "./types";
+
+const createRule = ESLintUtils.RuleCreator((name) => name);
+
+const isI18nNode = (node: Identifier) => {
+  return node.name === "i18n";
+};
+
+const hasTranslatePropertyIdentifier = (
+  node: any
+): node is JSXMemberExpression => {
+  return (
+    "parent" in node &&
+    "property" in node.parent &&
+    "name" in node.parent.property &&
+    node.parent.property.name === "t"
+  );
+};
+
+const hasInvalidVariableDeclaration = (node: Identifier) => {
+  return (
+    // Get 3rd parent node to check if it is a variable declaration
+    node.parent?.parent?.parent?.type === AST_NODE_TYPES.VariableDeclarator
```

Comment:
Why 3rd one? is it always the case, not dynamic?

Reply:
I made a dynamic one first by checking all parents nodes but it seems to be very long and bad in terms of performance when we run an `yarn lint`. So I decided to stay with this static checking which is working great because we just want to avoid this kind of usage `const key = i18n.t('key');`. And in that case the 3rd parent is the variable itself. If we detect a variable in the 3rd parent it means we're not fitting with the rule. Let me know if you have any other bad example in mind I can add them in the invalid statements.

Reply:
Here an example that should be invalid:
```
import i18n from '@locales/i18n';
const values = {key: i18n.t('home.setupAccount.title')};
```

Reply:
[Fixed now with this commit](https://github.com/GetLuko/eslint-react-native-guidelines/pull/6/commits/4b59b245cac94bb9f07654bf93041e8331f44cc8)